Dockerfile

```diff
@@ -0,0 +1,17 @@
+FROM python:3.12.3 as base
+RUN curl -sSL https://install.python-poetry.org | python3 -
+ENV PATH=$PATH:/root/.local/bin/
+COPY pyproject.toml poetry.toml /opt/todoapp/
+WORKDIR /opt/todoapp/
+RUN poetry install
+COPY todo_app /opt/todoapp/todo_app
+
+FROM base as production
+RUN poetry add gunicorn
```

Comment:
Rather than run this as part of the Dockerfile, I'd suggest running it in your host so that the version can be checked into source control through the `pyproject.toml` - even though Gunicorn doesn't work on Windows, we can still add it to the project

Reply:
Noted, I've made a note to self to add this as part of my next PR.